Server/ReasnAPI/ReasnAPI/Controllers/EventsController.cs

```diff
@@ -1,171 +1,307 @@
+        }
+
+        imageService.CreateImages(imageDtos);
+        return Ok();
+    }
+
+    [HttpPut]
+    [Authorize(Roles = "Admin, Organizer")]
+    [Route("{slug}/images")]
+    [ProducesResponseType(StatusCodes.Status200OK)]
+    public async Task<IActionResult> UpdateEventImage([FromRoute] string slug, [FromForm] List<IFormFile> images)
+    {
+        var user = userService.GetCurrentUser();
+        var @event = eventService.GetEventBySlug(slug);
+
+        if (@event.OrganizerId != user.Id && user.Role != UserRole.Admin)
+        {
+            return Forbid();
+        }
+
+        var imageDtos = new List<ImageDto>();
+
+        foreach (var formFile in images.Where(formFile => formFile.Length > 0))
+        {
+            using var ms = new MemoryStream();
+            await formFile.CopyToAsync(ms);
+            var fileBytes = ms.ToArray();
+
+            imageDtos.Add(new ImageDto
+            {
+                ObjectId = @event.Id,
+                ObjectType = ObjectType.Event,
+                ImageData = fileBytes
+            });
+        }
+
+        if (@event.Id != imageDtos[0].ObjectId)
+        {
+            return NotFound();
+        }
+
+        imageService.UpdateImagesForEvent(@event.Id, imageDtos);
+        return Ok();
     }
 
-    [HttpPost]
-    [Authorize(Roles = "Admin, Organizer")]
+    [HttpGet]
     [Route("{slug}/images")]
-    public IActionResult AddEventImage(string slug)
+    [ProducesResponseType<List<string>>(StatusCodes.Status200OK)]
+    public IActionResult GetEventImages([FromRoute] string slug)
     {
-        throw new NotImplementedException();
+        var @event = eventService.GetEventBySlug(slug);
+        var images = imageService.GetImagesByEventId(@event.Id);
+        var count = images.Count();
+        var stringList = new List<string>();
+        for (int i = 0; i < count; i++)
+        {
+            stringList.Add($"/api/v1/Events/{slug}/image/{i}");
+        }
+        return Ok(stringList);
     }
 
-    [HttpPut]
-    [Authorize(Roles = "Admin, Organizer")]
-    [Route("{slug}/images/{imageId:int}")]
-    public IActionResult UpdateEventImage(string slug, int imageId)
+    [HttpGet]
+    [Route("{slug}/image/{id:int}")]
+    [ProducesResponseType<FileContentResult>(StatusCodes.Status200OK)]
+    public IActionResult GetEventImage([FromRoute] string slug, [FromRoute] int id)
     {
-        throw new NotImplementedException();
-    }
+        var @event = eventService.GetEventBySlug(slug);
+        var images = imageService.GetImagesByEventId(@event.Id).ToList();
 
-    [HttpDelete]
-    [Authorize(Roles = "Admin, Organizer")]
-    [Route("{slug}/images/{imageId:int}")]
-    public IActionResult DeleteEventImage(string slug, int imageId)
-    {
-        throw new NotImplementedException();
+        if (images.Count <= id)
+        {
+            return NotFound();
+        }
+
+        var image = images[id];
+
+        return File(image.ImageData, $"image/jpeg");
```

Comment:
For that I would reccomend making an method in imageService for that, because if event have 9 images for example 10MB we get from DB 9 * 9 * 10, because for each image we get all images

Reply:
changed